src/Calculator/GraphingSettings.xaml.cpp

```diff
@@ -0,0 +1,220 @@
+}
+
+void GraphingSettings::TrigUnitModeClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    if (Degrees->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Degrees;
+    }
+
+    if (Radians->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Radians;
+    }
+
+    if (Gradians->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Grads;
+    }
+}
+
+void GraphingSettings::UpdateExtents(Platform::Object ^ sender)
+{
+    FrameworkElement ^ fe = (FrameworkElement ^) sender;
+    FrameworkElement ^ currentFocus = (FrameworkElement ^) FocusManager::GetFocusedElement();
+
+    bool thisControlHasFocus = (fe == currentFocus);
+    TextBox ^ tb = reinterpret_cast<TextBox ^>(sender);
+    bool widthChanged = false;
+    bool heightChanged = false;
+
+    try
+    {
+        double newVal = std::stod(tb->Text->Begin());
+        if (tb->Name == "x_min" && newVal != m_xMin)
+        {
+            widthChanged = true;
+            m_xMin = newVal;
+            m_ParentGrapher->XAxisMin = m_xMin;
+        }
+
+        if (tb->Name == "x_max" && newVal != m_xMax)
+        {
+            widthChanged = true;
+            m_xMax = newVal;
+            m_ParentGrapher->XAxisMax = m_xMax;
+        }
+
+        if (tb->Name == "y_min" && newVal != m_yMin)
+        {
+            heightChanged = true;
+            m_yMin = newVal;
+            m_ParentGrapher->YAxisMin = m_yMin;
+        }
+
+        if (tb->Name == "y_max" && newVal != m_yMax)
+        {
+            heightChanged = true;
+            m_yMax = newVal;
+            m_ParentGrapher->YAxisMax = m_yMax;
+        }
+
+        // Only the focused element get's to trigger the recompute
+
+        if (thisControlHasFocus)
+        {
+            if (m_preserveAspectRatio)
+            {
+                if (widthChanged)
+                {
+                    m_width = m_xMax - m_xMin;
+                    m_height = m_width / m_currentAspectRatio;
+                    double bounds = m_height / 2;
+                    m_yMin = -bounds;
+                    m_yMax = bounds;
+                    m_ParentGrapher->YAxisMin = m_yMin;
+                    m_ParentGrapher->YAxisMax = m_yMax;
+                }
+
+                if (heightChanged)
+                {
+                    m_height = m_yMax - m_yMin;
+                    m_width = m_height * m_currentAspectRatio;
+                    double bounds = m_width / 2;
+                    m_xMin = -bounds;
+                    m_xMax = bounds;
+                    m_ParentGrapher->XAxisMin = m_xMin;
+                    m_ParentGrapher->XAxisMax = m_xMax;
+                }
+
+                if (widthChanged || heightChanged)
+                {
+                    x_min->Text = m_xMin.ToString();
```

Comment:
I think this may add trailing zeroes, if it does, could we run this through NumberFormattingUtils::TrimTrailingZeros?